File: cpp/program/Laptop.cpp
#include "Perangkat.cpp"

// Class laptop (inherit from Perangkat)
class Laptop : public Perangkat
{
private:
    string cpu;
    string gpu;
    int ram;
    int storage;

public:
    // Constructor without param
    Laptop()
    {
        cpu = "";
        gpu = "";
        ram = 0;
        storage = 0;
    }

    // Constructor with parameter
    Laptop(string id, string nama, int harga, string merek, string kode_sku, int garansi, string cpu, string gpu, int ram, int storage)
    {
        setId(id);
        setNama(nama);
        setHarga(harga);
        setMerek(merek);
        setKodeSKU(kode_sku);
        setGaransi(garansi);
        this->cpu = cpu;
        this->gpu = gpu;
        this->ram = ram;
        this->storage = storage;
    }

    // Getter & setter for cpu
    void setCpu(string cpu){this->cpu = cpu;}
    string getCpu(){return cpu;}

    // Getter & setter for gpu
    void setGpu(string gpu){this->gpu = gpu;}
    string getGpu(){return gpu;}

    // Getter & setter for ram
    void setRam(int ram){this->ram = ram;}
    int getRam(){return ram;}

    // Getter & setter for storage
    void setStorage(int storage){this->storage = storage;}
    int getStorage(){return storage;}

    // Destructor
    ~Laptop(){}
};

File: php/program/Laptop.php
<?php

// Class Laptop extends Perangkat
class Laptop extends Perangkat {
    private $cpu;
    private $gpu;
    private $ram;
    private $storage;

    // Constructor
   public function __construct($id = "", $nama = "", $harga = 0,  $image = "", $merek = "", $kode_sku = "", $garansi = 0, $cpu = "", $gpu = "", $ram = 0, $storage = 0) {
        parent::__construct($id, $nama, $harga, $image, $merek, $kode_sku, $garansi);
        $this->cpu = $cpu;
        $this->gpu = $gpu;
        $this->ram = $ram;
        $this->storage = $storage;
    }

    // Getter & Setter for cpu, gpu, ram, storage
    public function setCpu($cpu) { $this->cpu = $cpu; }
    public function getCpu() { return $this->cpu; }

    public function setGpu($gpu) { $this->gpu = $gpu; }
    public function getGpu() { return $this->gpu; }

    public function setRam($ram) { $this->ram = $ram; }
    public function getRam() { return $this->ram; }

    public function setStorage($storage) { $this->storage = $storage; }
    public function getStorage() { return $this->storage; }
}
